src/main/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/RefreshOrchestrator.java

```diff
@@ -78,14 +81,20 @@ public void validate(Long jobId, UserRequest userRequest) {
             throw new BadRequestException("Invalid JobId request");
         }
 
-
         if (userRequest != null && nullCheckPredicate.test(userRequest.getUserIds())) {
-            //Extract and Validate received users List
+            // Extract and Validate received users List
             parseRequestService.validateUserRequest(userRequest);
             log.info("Validated userIds {}", userRequest.getUserIds());
         }
+
+        // Ensure only permitted service is invoking caseworker refresh API
+        final String serviceName = securityUtils.getServiceName();
+        if (!authorisedServices.contains(serviceName)) {
+            throw new BadRequestException("Invoking service is not permitted to call the Refresh API");
```

Comment:
Can we throw a 403 e.g. ForbiddenException.  and can we move the error message to a constant in one of:
* `src/main/java/uk/gov/hmcts/reform/orgrolemapping/v1/V1.java`
* `src/main/java/uk/gov/hmcts/reform/orgrolemapping/apihelper/Constants.java`
* `/src/main/java/uk/gov/hmcts/reform/orgrolemapping/controller/advice/ErrorConstants.java`

Reply:
Don't forget to update the new swagger messages.